### PR TITLE
Rocketeer support and updates to Drop/Massdrop keyboards

### LINKED ISSUE #6288: Add support for Massdrop x Zslane Mercury Rocketeer
Per the [QMK Docs](https://docs.qmk.fm/#/newbs_building_firmware_configurator?id=selecting-your-keyboard), I'm requesting firmware to be available for the [Massdrop x Zslane Mercury Rocketeer](https://drop.com/buy/64755).

Cheers!

### PR DESCRIPTION
**Pull request by just-another-jxliu**
## Description

New keyboards:
* Rocketeer support
* 1800-layout SHIFT prototype support, will need to change in the future to support the layout change to compact 1800.

For CTRL, ALT, Rocketeer, SHIFT:
* LED matrix: new toggle to only enable every other edge LED and new dedicated edge brightness controls.
* LED matrix: new pattern directions up/down and circular.
* Debounce behavior change.
* Increase soft boot minimum hold time from 0.5 seconds to 2 seconds
* Remove `U_T_AUTO`/`usb_extra_manual`
* Modified power management
* Under-the-hood changes to accommodate hardware differences between CTRL, ALT, Rocketeer, and SHIFT.

Keymap issues:
* `massdrop/ctrl:responsive_pattern` compiles, but LEDs are not functional because the keymap uses outdated LED functions. Same issue on master. Will fix in a future PR.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Closes #6288

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by just-another-jxliu** [comment]
I will add /default and /mac keymaps for Rocketeer to fix the CI failures.

**Comment by drashna** [comment]
Yeah, the keyboards need the `default` keymap, at a minimum. And Travis will fail without it. 

And glad to see this!

**Comment by newtonapple** [comment]
@just-another-jxliu do you plan to merge this soon?

**Comment by drashna** [comment]
This needs to be rebased with the latest master, since there was a large scale change in the formatting.

**Comment by drashna** [comment]
@just-another-jxliu any update on this?    Need any help with this? 

I ask, since I'd like to get this merged in sooner rather than later.

**Comment by Nathancooke7** [comment]
@just-another-jxliu do you plan to resolve the conflicts so this can be merged? I tried taking a look but it's beyond me.

**Comment by drashna** [comment]
Yes, please.  We'd like to get the merge conflicts fixed, and this PR in sooner rather than later.  

Also, tagging @patrickmt just in case.

**Comment by pschmuckal** [comment]
Resolved merge issues, however clang changes not merged but as I understand Travis will handle that.  Added missing default & mac keymaps.

Should be ready to go.

**Comment by drashna** [comment]
@pschmuckal if you have the repo up to date, you can always run `bin/qmk cformat` and that should manually run clang-format on the repo.   Just for a heads up.

Also, do you know of the best way to get ahold of whomever for the massdrop code?  

Namely, we would like to see any updates merged in QMK more rapidly

**Comment by skullydazed** [comment]
We've just been talking internally about this PR. There is going to be a problem getting the necessary approvals to merge this given the large size. In general we strongly prefer to receive PR's that are limited in scope and are comprised of 1 or 2 atomic units- not just a code dump from months worth of development.

There are a lot of violation of DRY that I am not a fan of approving, personally. It makes it difficult to truly asses the impact of the code and makes it harder for anyone trying to understand how this works. In addition, it seems like the use of custom, keyboard specific code is increasing with this PR. We would strongly prefer to see that moving in the other direction.

The fastest way to get everything in would be to close this PR and submit small, focused PR's instead. Otherwise, you will have to give us some time to get through this one.

**Comment by drashna** [comment]
There are a number of merge conflicts here.

**Comment by pschmuckal** [comment]
What do you need from us (Drop) to get this PR closed?

We realize it is not ideal due to # of mods, but these changes were done almost a yr ago by a contractor no longer involved and are needed to support our shipping products.

Also heads up - there are a number of additional changes coming down the pipe to support other new products coming very soon.

**Comment by drashna** [comment]
@pschmuckal more time.  It's a large change, touching a BUNCH of files.   

What would help in the future is smaller, more incremental changes, as they're easier to review and get merged in, in a timely manner.

File: keyboards/massdrop/rocketeer/config.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID           0x04D8
+#define PRODUCT_ID          0xEE66
```
Comment (drashna) [inline keyboards/massdrop/rocketeer/config.h]: These are used already, and by Microchip.

File: keyboards/massdrop/ctrl/matrix.c [hunk 2]
```diff
@@ -80,27 +83,29 @@ void matrix_init(void)
 }
 
 uint64_t mdebouncing = 0;
+bool debouncing = false;
+
 uint8_t matrix_scan(void)
```
Comment (drashna) [inline keyboards/massdrop/ctrl/matrix.c]: This doesn't need to be done now, but would really like to see the debouncing API integrated into the matrix here: 
https://docs.qmk.fm/#/feature_debounce_type

File: keyboards/massdrop/rocketeer/config.h (hunk 1 above)
Comment (drashna) [inline keyboards/massdrop/rocketeer/config.h]: These belong to Microchip.

Comment (pschmuckal) [reply keyboards/massdrop/rocketeer/config.h]: Yes, we have a sublicense with Microchip to use their VID & PIDs.  All existing Drop keyboards use Microchip VID.

**Comment by pschmuckal** [comment]
@drashna Understood, and apologize once again about the excessive number of mods here.  Note that our HW design has changed significantly for newer products which is one reason why so many files were touched, as well as adding additional 'config' files into the keyboards dirs as well.  Not sure I could split up this PR easily and still be buildable.

**Comment by drashna** [comment]
Some merge conflicts here

**Comment by drashna** [comment]
@just-another-jxliu @pschmuckal was there any plans to update this and resolve the merge conflicts here?

**Comment by tzarc** [comment]
Apparently the breaking change label is preventing stalebot from closing this PR.
Please refer to #11119. Closing.

**Pull request closed without merging**